src/common/m_phase_change.f90

```diff
@@ -0,0 +1,2098 @@
+        
+            ! calculating initial estimate for pressure in the pT-relaxation procedure. I will also use this variable to 
+            ! iterate over the Newton's solver
+            pO = 0.0D0
+        
+            ! Maybe improve this condition afterwards. As long as the initial guess is in between -min(p_inf)
+            ! and infinity, a solution should be able to be found.
+            pS = MAX( MAXVAL( pk ), -1.0D0 * MINVAL( p_infA ) ) + 1.0D4
+        
+            ! Newton Solver for the pT-equilibrium
+            ns = 0
+            DO WHILE ( ( DABS( pS - pO ) .GT. palpha_eps ) .AND. ( DABS( ( pS - pO ) / pO ) .GT. palpha_eps ) &
+                        .OR. ( ns .EQ. 0 ) )
+
+                ! increasing counter
+                ns = ns + 1
+        
+                ! updating old pressure 
+                pO = pS
+        
+                ! updating functions used in the Newton's solver
+                gpp = 0.0D0
+                gp = 0.0D0
+                hp = 0.0D0
+
+                DO i = 1, num_fluids
+
+                    ! given pS always change, I need ig( i ) and gp to be in here, as it dynamically updates.
+                    ! not that I do not need to use p_infA here, but I will do it for consistency
+                    IF ( i .NE. ig( i ) ) THEN
+
+                        gp =   gp + ( g_min( i ) - 1.0D0 ) & 
+                        * q_cons_vf( i + cont_idx%beg - 1 )%sf( j, k, l ) * cv( i ) &
+                        * ( rhoe + pS - mQ ) / ( mCP * ( pS + p_infA( i ) ) )
+            
+                        gpp = gpp + ( g_min( i ) - 1.0D0 ) & 
+                        * q_cons_vf( i + cont_idx%beg - 1 )%sf( j, k, l ) * cv( i ) &
+                        * ( p_infA( i ) - rhoe + mQ ) / ( mCP * ( pS + p_infA( i ) ) ** 2 )
+
+                    END IF
+
+                END DO
+        
+                hp = 1.0D0 / ( rhoe + pS - mQ ) + 1.0D0 / ( pS + MINVAL( p_infA ) )
+        
+                ! updating common pressure for the newton solver
+                pS = pO + ( ( 1.0D0 - gp ) / gpp ) / ( 1.0D0 - ( 1.0D0 - gp + DABS( 1.0D0 - gp ) ) &
+                / ( 2.0D0 * gpp ) * hp )
+        
+                ! check if solution is out of bounds (which I believe it won`t happen given the solver is gloabally convergent.
+                ! I am not an expert in numerical methods though)
+                IF ( pS .LE. -1.0D0 * MINVAL( p_infA ) ) THEN
+
+                    CALL s_tattletale( (/ 0.0D0, 0.0D0 /), RESHAPE( (/ 0.0D0, 0.0D0, 0.0D0, 0.0D0 /), (/ 2, 2 /) ) &
+                    , j, (/ 0.0D0, 0.0D0, 0.0D0, 0.0D0 /), k, l, mQ, p_infA, pS, qv, (/ pS - pO, pS + pO /) &
+                    , rhoe, q_cons_vf, TS )
+    
+                    CALL s_mpi_abort('Solver for the pT-relaxation solver failed (m_phase_change, s_infinite_pt_relaxation_k) &
+                    . Please, check the pressure value. Aborting!')
+                
+                ELSEIF ( ISNAN( pS ) ) THEN
```

Comment:
`isnan` will fail on NVHPC. `ieee_is_nan` from `ieee_arithmetic` might resolve this.

Reply:
`./mfc.sh format` run
`isnan` changed to `ieee_is_nan`

`m_phase_change.f90` commited, with non-[no ci], as per request